src/main/java/org/dcsa/tnt/service/impl/GenericEventServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.dcsa.tnt.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.dcsa.core.events.model.EquipmentEvent;
+import org.dcsa.core.events.model.Event;
+import org.dcsa.core.events.model.ShipmentEvent;
+import org.dcsa.core.events.model.TransportEvent;
+import org.dcsa.core.events.repository.EventRepository;
+import org.dcsa.core.events.service.EquipmentEventService;
+import org.dcsa.core.events.service.GenericEventService;
+import org.dcsa.core.events.service.ShipmentEventService;
+import org.dcsa.core.events.service.TransportEventService;
+import org.dcsa.core.exception.NotFoundException;
+import org.dcsa.core.extendedrequest.ExtendedRequest;
+import org.dcsa.core.service.impl.ExtendedBaseServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class GenericEventServiceImpl extends ExtendedBaseServiceImpl<EventRepository, Event, UUID> implements GenericEventService {
+
+    @Autowired
```

Comment:
Since you are using `@RequiredArgsConstructor` you do not need to use `@Autowired`, just declare the fields as private final and it will use constructor injection.

Reply:
Seems like this does not work. If I remove the `AutoWired` tag in e.g. `TNTEventServiceImpl` the properties are `null`.